GDJS/Runtime/pixi-renderers/pixi-atlas-manager.ts

```diff
@@ -0,0 +1,191 @@
+        this.load(textResources[i].name, onLoad);
+      }
+    }
+
+    /**
+     * Request the text file from the given resource name.
+     * This method is asynchronous. When loaded, the `callback` is called with the error
+     * (null if none) and the loaded test (a string).
+     *
+     * @param resourceName The resource pointing to the json file to load.
+     * @param callback The callback function called when json is loaded (or an error occurred).
+     */
+    load(resourceName: string, callback: AtlasManagerRequestCallback): void {
+      const resource = this._resources.find((resource) => isTextResource(resource) && resource.name === resourceName);
+      if (!resource) {
+        return callback(new Error(`Can't find resource with name: "${resourceName}" (or is not a text resource).`), null);
+      }
+
+      // Don't fetch again an object that is already in memory
+      if (this.isLoaded(resourceName)) {
+        return callback(null, this._loadedAtlases[resourceName]);
+      }
+
+      // Don't fetch again an object that is already being fetched.
+      const callbacks = this._callbacks[resourceName];
+      if (callbacks) {
+        callbacks.push(callback);
+        return;
+      }
+
+      this._callbacks[resourceName] = [callback];
+      const xhr = new XMLHttpRequest();
+      xhr.responseType = 'text';
+      xhr.withCredentials = this._resourcesLoader.checkIfCredentialsRequired(resource.file);
+      xhr.open('GET', this._resourcesLoader.getFullUrl(resource.file));
+      xhr.onload = () => {
+        if (xhr.status !== 200) {
+          this.callCallback(resourceName, `HTTP error: ${xhr.status} (${xhr.statusText })`, null);
+        } else {
+          this._loadedAtlases[resourceName] = xhr.response;
+
+          new PIXI_SPINE.TextureAtlas(
+            xhr.response,
+            (path, textureCb) => {
+              const atlasImagePath = resourceName.match(/(^.*[\\\/]|^[^\\\/].*)/i)![0] + path;
+              textureCb(this._imageManager.getPIXITexture(atlasImagePath).baseTexture)
+            },
+            (atlas) => {
+              this._loadedTextureAtlases[resourceName] = atlas;
+              this.callCallback(resourceName, null, xhr.response);
+            });
+        }
+      };
+      xhr.onerror = () => this.callCallback(resourceName, 'Network error', null);
+      xhr.onabort = () => this.callCallback(resourceName, 'Request aborted', null);
+      xhr.send();
+    }
+
+    protected callCallback(resourceName: string, errorMessage: string, text: null): void
+    protected callCallback(resourceName: string, errorMessage: null, text: string): void
+    protected callCallback(resourceName: string, errorMessage: string | null, text: string | null): void {
+      if (!this._callbacks[resourceName]) { return; }
+  
+      for (const callback of this._callbacks[resourceName]) {
+        const error = typeof errorMessage === 'string' ? new Error(errorMessage) : null;
+        callback(error, text);
```

Comment:
Add non null assertion for text

Reply:
done

Reply:
I meant simple not-null check or asset utility function that throws only in dev mode